textdb/textdb-dataflow/src/main/java/com/google/re2j/PublicRegexp.java

```diff
@@ -0,0 +1,225 @@
+	    CHAR_CLASS,         // Matches Runes interpreted as range pair list
+	    ANY_CHAR_NOT_NL,    // Matches any character except '\n'
+	    ANY_CHAR,           // Matches any character
+	    BEGIN_LINE,         // Matches empty string at end of line
+	    END_LINE,           // Matches empty string at end of line
+	    BEGIN_TEXT,         // Matches empty string at beginning of text
+	    END_TEXT,           // Matches empty string at end of text
+	    WORD_BOUNDARY,      // Matches word boundary `\b`
+	    NO_WORD_BOUNDARY,   // Matches word non-boundary `\B`
+	    CAPTURE,            // Capturing subexpr with index cap, optional name name
+	    STAR,               // Matches subs[0] zero or more times.
+	    PLUS,               // Matches subs[0] one or more times.
+	    QUEST,              // Matches subs[0] zero or one times.
+	    REPEAT,             // Matches subs[0] [min, max] times; max=-1 => no limit.
+	    CONCAT,             // Matches concatenation of subs[]
+	    ALTERNATE,          // Matches union of subs[]
+	    
+	    // Pseudo ops, used internally by Parser for parsing stack:
+	    // These shouldn't be in the final parse tree
+	    LEFT_PAREN,
+	    VERTICAL_BAR;
+	}
+	
+	/**
+	 * op's enum type Regexp.Op is not public
+	 * this converts op to an equivalent enum type, PublicOp
+	 * then returns it
+	 * 
+	 * @return PublicRegex.PublicOp, an enum type representing the operator
+	 */
+	public PublicOp getOp() {
+		try {
+			return PublicOp.valueOf(this.op.toString());
+		} catch (IllegalArgumentException e) {
+			return PublicOp.NO_MATCH;
+		}
+		
+	}
+	
+	/**
+	 * flags is a bitmap of parse flags
+	 * @see PublicRE2 for possible flags
+	 * @return a bitmap of parse flags
+	 */
+	public int getFlags() {
+		return this.flags;
+	}
+	
+	/**
+	 * @return an array of subexpressions with type PublicRegexp
+	 */
+	public PublicRegexp[] getSubs() {
+		return this.publicSubs;
+	}
+	
+	/**
+	 * runes: a sequence of characters
+	 * it stores information related to literals and character classes
+	 * it has different interpretations for different ops
+	 * for example,
+	 * regex: "[a-z]", runes: [a,z]
+	 * 		interpretation: a character class from a to z
+	 * regex: "[a-cx-z]", runes: [a,c,x,z]
+	 * 		interpretation: a character class from a to c, and from x to z
+	 * regex: "cat", runes [c,a,t]
+	 * 		interpretation: a literal "cat"
+	 * 
+	 * @return an array of runes
+	 */
+	public int[] getRunes() {
+		return this.runes;
+	}
+	
+	/**
+	 * min and max are used for repetitions
+	 * for example,
+	 * regex: a{3,5}, min will be 3, max will be 5
+	 * @return int indicating minimum number of repetitions
+	 */
+	public int getMin() {
+		return this.min;
+	}
+	
+	/**
+	 * min and max are used for repetitions
+	 * for example,
+	 * regex: a{3,5}, min will be 3, max will be 5
+	 * @return int indicating maxinum number of repetitions
+	 */
+	public int getMax() {
+		return this.max;
+	}
+	
+	/**
+	 * cap is the capturing index
```

Comment:
Change `cap` to `"cap"` (with the quotes).  Treat it as a general comment and make other changes as well.